modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/MethodArgumentResolverDelegate.java

```diff
@@ -336,28 +326,26 @@ private void initArgumentResolvers() {
     return parameterValues;
   }
 
-  private ArgumentResolver<Object> addResolverDecorators(ArgumentResolver<Object> resolver,
-                                                         CursorComponentDecoratorFactory componentDecoratorFactory,
-                                                         Parameter parameter) {
+  private ArgumentResolver<Object> addResolverDecorators(ArgumentResolver<Object> resolver, Parameter parameter) {
     Class<?> argumentType = parameter.getType();
     if (argumentType.isPrimitive()) {
       resolver = addPrimitiveTypeDefaultValueDecorator(resolver, argumentType);
     } else if (InputStream.class.equals(argumentType)) {
-      resolver = new InputStreamArgumentResolverDecorator(resolver, componentDecoratorFactory);
+      resolver = new InputStreamArgumentResolverDecorator(resolver);
     } else if (TypedValue.class.equals(argumentType)) {
       if (parameter.getParameterizedType() instanceof ParameterizedType) {
         Type generic = ((ParameterizedType) parameter.getParameterizedType()).getActualTypeArguments()[0];
         if (generic instanceof Class) {
           Class<?> genericClass = (Class<?>) generic;
-          if (CursorProvider.class.isAssignableFrom(genericClass) || Object.class.equals(genericClass)) {
-            resolver = new TypedValueCursorArgumentResolverDecorator(resolver, componentDecoratorFactory);
-          } else {
-            resolver = new TypedValueArgumentResolverDecorator(resolver, componentDecoratorFactory);
+          if (CursorProvider.class.isAssignableFrom(genericClass)
+              || InputStream.class.isAssignableFrom(genericClass)
```

Comment:
Had to add the `InputStream` to the classes that will be decorated with `TypedValueCursorArgumentResolverDecorator` because we want to decorate with `UnclosableCursorStream` (see #11209). Previously we had the `TypedValueArgumentResolverDecorator`, but I think we don't need to have a different one now, because it will be doing exactly the same thing, and also, we only need to care about `InputStreams` and not everything else. But please check if I'm mistaken.

Reply:
@ndinu @dabreu  please review this

Reply:
👍